source/FreeRTOS_UDP_IP.c

```diff
@@ -94,33 +79,23 @@ UDPPacketHeader_t xDefaultPartUDPPacketHeader =
  */
 void vProcessGeneratedUDPPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer )
 {
-    const UDPPacket_t * pxUDPPacket;
-
-    if( pxNetworkBuffer != NULL )
+    switch( usFrameType( pxNetworkBuffer ) )
     {
-        /* Map the UDP packet onto the start of the frame. */
-
-        /* MISRA Ref 11.3.1 [Misaligned access] */
-        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
-        /* coverity[misra_c_2012_rule_11_3_violation] */
-        pxUDPPacket = ( ( UDPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
-
-        switch( pxUDPPacket->xEthernetHeader.usFrameType )
-        {
-            #if ( ipconfigUSE_IPv4 != 0 )
-                case ipIPv4_FRAME_TYPE:
-                    vProcessGeneratedUDPPacket_IPv4( pxNetworkBuffer );
-                    break;
-            #endif
-            #if ( ipconfigUSE_IPv6 != 0 )
-                case ipIPv6_FRAME_TYPE:
-                    vProcessGeneratedUDPPacket_IPv6( pxNetworkBuffer );
-                    break;
-            #endif
-            default:
-                /* do nothing, coverity happy */
+        #if ( ipconfigUSE_IPv4 != 0 )
+            case ipIPv4_FRAME_TYPE:
+                vProcessGeneratedUDPPacket_IPv4( pxNetworkBuffer );
+                break;
+        #endif
+        #if ( ipconfigUSE_IPv6 != 0 )
+            case ipIPv6_FRAME_TYPE:
+                vProcessGeneratedUDPPacket_IPv6( pxNetworkBuffer );
                 break;
-        }
+        #endif
+        default:
+            /* Must Know Frame Type first before sending here */
+            FreeRTOS_debug_printf( ( "vProcessGeneratedUDPPacket: Unsupported Packet\n" ) );
+            configASSERT( pdFALSE );
```

Comment:
I think I was expecting all invalid packets to have been ignored already somewhere in `prvProcessEthernetPacket`, `prvProcessIPPacket`, `prvAllowIPPacketIPv6`, or `prvAllowIPPacketIPv4`, in which case it would never reach here.